Lib/typing.py

```diff
@@ -1674,7 +1674,7 @@ def __class_getitem__(cls, params):
                             "with at least two arguments (a type and an "
                             "annotation).")
         msg = "Annotated[t, ...]: t must be a type."
-        origin = _type_check(params[0], msg)
+        origin = _type_check(params[0], msg, is_class=True)
```

Comment:
This is a bit of an abuse of the `is_class` argument, maybe we should rename it (suggestion: `allow_special_forms`).

Also, this needs tests in `test_typing.py`.

Reply:
I went ahead and changed it to your suggestion and added a test for special form containment.